lib/WIFI/WIFI.cpp

```diff
@@ -0,0 +1,299 @@
+    for (i = 1; i <= password.length(); i++)
+    {
+        EEPROM.write(NVM_PASSWORD_ADDRESS + i, password[i - 1]);
+    }
+    EEPROM.write(NVM_PASSWORD_ADDRESS + i, ';');
+
+    if (EEPROM.commit())
+    {
+        Serial.println("EEPROM successfully committed");
+    }
+    else
+    {
+        Serial.println("ERROR! EEPROM commit failed");
+    }
+
+    if (true)
+    {
+        Serial.println("Credentials Received");
+        success = true;
+    }
+    return success;
+}
+
+/**************************************************************************************************/
+
+/**
+ *  Returns the IP Address of the System.
+ * 
+ *  @return local IP.
+ */
+const IPAddress &WIFI::getIPAddress(void)
+{
+    return localIP;
+}
+
+/* PROTECTED METHODES ****************************************************************************/
+
+/* PRIVATE METHODES ******************************************************************************/
+
+/**************************************************************************************************/
+
+/**
+ *  Connect to Wireless Access Point.
+ * 
+ *  @return success.
+ */
+bool WIFI::connectStation()
+{
+    bool success = false;
+
+    unsigned long startAttempTime = millis();
+
+    Serial.println("Connecting to \"" + STA_SSID + "\"...");
+    while ((WiFi.status() != WL_CONNECTED) && ((millis() - startAttempTime) < WIFI_TIMEOUT_MS))
+    {
+        delay(100);
+    }
+
+    if (WL_CONNECTED == WiFi.status())
+    {
+        Serial.println("Connected Succesfully.");
+        success = true;
+        localIP = WiFi.localIP();
+    }
+
+    return success;
+}
+
+/**************************************************************************************************/
+
+/**
+ *  Imports Credentials from EEPROM.
+ * 
+ *  @return success.
+ */
+bool WIFI::importCredentials()
+{
+    bool success = false;
+
+    if (';' == char(EEPROM.read(NVM_SSID_ADDRESS)))
+    {
+        char temp;
+
+        for (uint8_t i = 1; i < CREDENTIALS_MAX_LENGTH; i++)
+        {
+            temp = EEPROM.read(NVM_SSID_ADDRESS + i);
+            if (';' != temp)
+            {
+                STA_SSID += temp;
+            }
+            else
+            {
+                break;
+            }
+        }
+
+        if (';' == char(EEPROM.read(NVM_PASSWORD_ADDRESS)))
+        {
+            for (uint8_t i = 1; i < CREDENTIALS_MAX_LENGTH; i++)
+            {
+                temp = EEPROM.read(NVM_PASSWORD_ADDRESS + i);
+                if (';' != temp)
+                {
+                    STA_PASSWORD += temp;
+                }
+                else
+                {
+                    break;
+                }
+            }
+        }
+
+        success = true;
+    }
+
+    return success;
+}
+
+/**************************************************************************************************/
+
+/**
+ *  Deletes stored Credentials.
+ */
+void WIFI::clearEEPROM()
+{
+    for (int i = 0; i < 512; i++)
```

Comment:
512 = Magic number and used several times. Please use a static constant variable!